fe/fe-core/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java

```diff
@@ -1011,13 +912,17 @@ public Database getDatabase(String catalogName, String databaseName)
     @Override
     public List<Table> getTableObjectsByName(String dbName, List<String> tableNames)
             throws MetaException, InvalidOperationException, UnknownDBException, TException {
-        throw new TException("method not implemented");
+        return getTableObjectsByName(null, dbName, tableNames);
     }
 
     @Override
     public List<Table> getTableObjectsByName(String catName, String dbName, List<String> tableNames)
             throws MetaException, InvalidOperationException, UnknownDBException, TException {
-        throw new TException("method not implemented");
+        GetTablesRequest req = new GetTablesRequest(dbName);
+        req.setCatName(catName);
+        req.setTblNames(tableNames);
+        req.setCapabilities(version);
+        return client.get_table_objects_by_name_req(req).getTables();
```

Comment:
Is it interface compatible with older HMS versions?

Reply:
I'm not sure, I haven't an older hms version to have a test.

Reply:
maybe you need to test if you can access 1.x/2.x/3.x hms

Reply:
I am using lower API instead, it already has in hive-1.xxx.

https://github.com/apache/hive/blob/branch-1/metastore/if/hive_metastore.thrift